transport.go

```diff
@@ -183,17 +191,25 @@ func (t *transport) Dial(ctx context.Context, raddr ma.Multiaddr, p peer.ID) (tp
 		return nil, err
 	}
 	tlsConf, keyCh := t.identity.ConfigForPeer(p)
-
-	if ok, isClient, _ := n.GetSimultaneousConnect(ctx); ok && !isClient {
-		return t.holePunch(ctx, network, addr, p)
+	if ok, isClient, _ := network.GetSimultaneousConnect(ctx); ok && !isClient {
+		return t.holePunch(ctx, netw, addr, p)
 	}
 
-	pconn, err := t.connManager.Dial(network, addr)
+	scope, err := t.rcmgr.OpenConnection(network.DirOutbound, false)
```

Comment:
and here too, where is the scope passed?

Reply:
Fixed by using the new `transport.CapableConn` interface.